src/include/ikarus/Assembler/SimpleAssemblers.h

```diff
@@ -23,75 +27,76 @@ namespace Ikarus::Assembler {
   private:
     double& getScalarImpl(Ikarus::FiniteElements::ScalarAffordances scalarAffordances) {
       scal = 0;
-      for (auto [fe, dofs, vars] : dofManager_->elementDofsVariableTuple())
-        scal += calculateScalar(fe, vars, scalarAffordances);
+      for (auto [fe, dofs, vars] : feManager_->elementDofsVariableTuple()) {
+        scal += calculateScalar(fe, scalarAffordances, vars);
+      }
       return scal;
     }
 
   private:
-    DofManagerType* dofManager_;
+    DofManagerType* feManager_;
     double scal{};
   };
 
   template <typename DofManagerType>
   class VectorAssembler {
   public:
     explicit VectorAssembler(DofManagerType& dofManager)
-        : dofManager_{&dofManager}, vec{Eigen::VectorXd::Zero(dofManager_->correctionSize())} {}
+        : feManager_{&dofManager}, vec{Eigen::VectorXd::Zero(feManager_->correctionSize())} {}
 
     Eigen::VectorXd& getVector(Ikarus::FiniteElements::VectorAffordances vectorAffordances) {
       return getVectorImpl(vectorAffordances);
     }
 
   private:
     Eigen::VectorXd& getVectorImpl(Ikarus::FiniteElements::VectorAffordances vecAffordances) {
-      vec.setZero(dofManager_->correctionSize());
-      for (auto [fe, dofIndices, vars] : dofManager_->elementDofsVariableTuple()) {
-        assert(dofIndices.size() == calculateVector(fe, vars, vecAffordances).size()
+      vec.setZero(feManager_->correctionSize());
+      for (auto&& [fe, dofIndices, vars] : feManager_->elementDofsVariableTuple()) {
+        assert(dofIndices.size() == calculateVector(fe, vecAffordances, vars).size()
                && "The returned vector has wrong rowSize!");
-        vec(dofIndices) += calculateVector(fe, vars, vecAffordances);
+        vec(dofIndices) += calculateVector(fe, vecAffordances, vars);

```

Comment:
calculateVector wird hier doppelt aufgerufen, erst im assert und dann nochmal in Zeile 57. Es ist wahrscheinlich effizienzer, calculateVector aufzurufen, das Ergebnis temporär zu speichern, dann den assert machen und dann Zeile 57 mit dem zwischengespeicherten Ergebnis als rechte Seite

Reply:
Gilt auch für Matrix Assembler

Reply:
Jo da wollte ich aber den release code nicht nerven mit meinen asserts. Im Release wird der assert ja komplett rausgeworfen. Wenn ich aber das ergebnis noch zuert speicher, dann taucht das auch im release code auf.